client/src/AddFriend.js

```diff
@@ -97,14 +103,40 @@ class AddFriend extends React.Component {
             onChange={this.handleChange}
             placeholder='Friends email'
           />
-          <input type='submit' value='Add Friend' />
+          {!this.state.isloading && submitButton()}
+          {this.state.isloading && loadButton()}
         </form>
-
+            
         <h5>Total Count</h5>
         <h5>0</h5>
       </div>
     )
   }
 }
 
+function loadButton() {
+  return (
+    <>
+  <Button variant="secondary" disabled>
+    <Spinner
+      as="span"
+      animation="border"
+      size="sm"
+      role="status"
+      aria-hidden="true"
+    />
+    Loading...
+  </Button>
+</>
+  
+  );
+}
+function submitButton() {
+  return (
+    <input type='submit' value='Add Friend' />
+  )
+}
```

Comment:
Notice what you've done -- you've separated out two smaller components (submit button / load button) from a larger one (AddFriend page). This is usually where you split them up, so that every component has its own file. The file structure might look like this:

```
/src
 - App.js (this is the base file, which routes to the different pages)
 - /AddFriend (new folder for each page of the app)
   - AddFriend.js (the actual page)
   - SubmitButton.js (a component on the page)
   - LoadingButton.js
 - /Login
   - Login.js

Reply:
This can be done in a later restructuring/cleanup MR though.